### PR TITLE
Sohu fix

### PR DESCRIPTION
**Pull request by yan12125**
Sohu info extractor currently fails from time to time. I fix it and add some test cases.

### CODE REVIEW

File: youtube_dl/extractor/sohu.py [hunk 1]
```diff
@@ -77,6 +104,11 @@ def _fetch_data(vid_id, mytv=False):
                     % (format_id, i + 1, part_count))
 
                 part_info = part_str.split('|')
+
+                # Sanitize URL to prevent download failure
+                if part_info[0][-1] == '/' and su[i][0] == '/':
+                    su[i] = su[i][1:]
+
                 video_url = '%s%s?key=%s' % (part_info[0], su[i], part_info[3])
```
Comment (jaimeMF) [inline youtube_dl/extractor/sohu.py]: I think the proper way to do that is `'%s?key=%s'  % (compat_urlparse.urljoin(part_info[0], su[i]), part_info[3])`.

Comment (yan12125) [inline youtube_dl/extractor/sohu.py]: It does not work. For example, when ```part_info[0]``` is ```http://70.39.188.10/sohu/s26h23eab6/2/``` and su[i] is ```/0/191/bymuey1CRhqd9706LeW3YD.mp4```, I need ```http://70.39.188.10/sohu/s26h23eab6/2/0/191/bymuey1CRhqd9706LeW3YD.mp4``` but I get ```http://70.39.188.10/0/191/bymuey1CRhqd9706LeW3YD.mp4```. The result is 404 not found.

**Comment by yan12125** [comment]
Add a more generic function to youtube_dl/utils.py based on the implementation at http://stackoverflow.com/a/8926421/3786245

**Comment by yan12125** [comment]
I've fixed title extraction (the original approach was unable to handle titles like "youtube-dl testing video") and add some test cases. Please review it.

**Comment by dstftw** [comment]
Thanks.

**Pull request merged**